CRM/Contact/Form/DedupeFind.php

```diff
@@ -51,6 +51,7 @@ public function buildQuickForm() {
     $groupList = array('' => ts('- All Contacts -')) + CRM_Core_PseudoConstant::nestedGroup();
 
     $this->add('select', 'group_id', ts('Select Group'), $groupList, FALSE, array('class' => 'crm-select2 huge'));
+    $this->add('text', 'limit', ts('No of contacts to find matches for '));
```

Comment:
OK, but I think that makes it more complicated from the UI point of view, and more confusing for end users. If you just increase the limit on the UI then you're searching the same people again and again, since you can't specify the range to search. I agree that limiting is useful, but I'm not sure how this solution actually improves anything in the UI since you'd have to select the first 1000 contacts in the DB to look for duplicates for. And then the first 2000, etc. etc. You still end up having to do the 'massive' query to search the entire contact base for duplicates.

At least I'd recommend having a 'no limit' option which is selected by default.